src/utilities/monorail/hooks.tsx

```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import {MonorailContext} from './context';
+
+export function useMonorail() {
+  return React.useContext(MonorailContext);
```

Comment:
Check if the result returns a value, and if not return a noop function. That will ensure this function always returns a function. that way we can avoid having to check if the context has a value in https://github.com/Shopify/polaris-react/pull/2368/files#diff-ac192d85916a4a5bf777ae9cb10c1a08R29